app/controllers/matches_controller.rb

```diff
@@ -1,10 +1,15 @@
 class MatchesController < ApplicationController
   def create
+    match_params = params[:match]
     date_time = DateTime.strptime(match_params[:date]+ " " + match_params[:time], "%Y-%m-%d %H:%M")
-    loser = ([match_params[:playerA], match_params[:playerB]] - [match_params[:winner]]).first
-    match = Match.create(played_at: date_time, participants: 2)
-    Result.create(match_id: match.id, user_id: match_params[:winner], place: 1)
-    Result.create(match_id: match.id, user_id: loser, place: 2)
+    match = Match.create(played_at: date_time, participants: match_params[:participants].to_i, event_id: match_params[:event])
+    
+    place = 1
+    while params[:match][place.to_s] != nil do
+      Result.create(match_id: match.id, user_id: match_params[place.to_s], place: place)
+      place += 1
+    end
```

Comment:
This is clunky... but it works. Matches are logged based on a "place" so the "winner" would be the player corersponding to the id in field `match[1]`. For Head to Head matches, the "loser" will always be in 2nd place, so `match[2]`. For multiplayer this allows us to log several results associated to a match and have several placements (no ties though)